src/Xamarin.Forms.PancakeView.Multi/Platforms/iOS/PancakeViewRenderer.cs

```diff
@@ -14,444 +14,436 @@
+			layer.ShadowColor = UIColor.Black.CGColor;
+			layer.ShadowOpacity = 0.24f;
+			layer.ShadowOffset = new CGSize(0, elevation);
+
+			if (pancake.Sides != 4)
+			{
+				layer.ShadowPath = ShapeUtils.CreatePolygonPath(bounds, pancake.Sides, pancake.CornerRadius.TopLeft, pancake.OffsetAngle).CGPath;
+			}
+			else
+			{
+				layer.ShadowPath = UIBezierPath.FromRoundedRect(bounds, cornerRadius).CGPath;
+			}
+
+			layer.MasksToBounds = false;
+		}
+
+		private CAGradientLayer CreateGradientLayer(int angle, CGRect rect)
+		{
+			var totalAngle = angle / 360.0;
+
+			// Calculate the new positions based on angle between 0-360.
+			var a = Math.Pow(Math.Sin(2 * Math.PI * ((totalAngle + 0.75) / 2)), 2);
+			var b = Math.Pow(Math.Sin(2 * Math.PI * ((totalAngle + 0.0) / 2)), 2);
+			var c = Math.Pow(Math.Sin(2 * Math.PI * ((totalAngle + 0.25) / 2)), 2);
+			var d = Math.Pow(Math.Sin(2 * Math.PI * ((totalAngle + 0.5) / 2)), 2);
+
+			// Create a gradient layer that draws our background.
+			return new CAGradientLayer
+			{
+				Frame = rect,
+				StartPoint = new CGPoint(1 - a, b),
+				EndPoint = new CGPoint(1 - c, d)
+			};
+		}
+
+		private CGPath CreateCornerPath(double topLeft, double topRight, double bottomRight, double bottomLeft, CGRect insetBounds)
+		{
+			var cornerPath = new CGPath();
+
+			// Start of our path is where the top left horizontal starts.
+			cornerPath.MoveToPoint(new CGPoint(topLeft + insetBounds.X, insetBounds.Y));
+
+			// Top line + top right corner
+			cornerPath.AddLineToPoint(new CGPoint(insetBounds.Width - topRight, insetBounds.Y));
+			cornerPath.AddArc((float)(insetBounds.X + insetBounds.Width - topRight), (float)(insetBounds.Y + topRight), (float)topRight, (float)(Math.PI * 1.5), (float)Math.PI * 2, false);
+
+			// Right side + bottom right corner
+			cornerPath.AddLineToPoint(insetBounds.Width + insetBounds.X, (float)(insetBounds.Height - bottomRight));
+			cornerPath.AddArc((float)(insetBounds.X + insetBounds.Width - bottomRight), (float)(insetBounds.Y + insetBounds.Height - bottomRight), (float)bottomRight, 0, (float)(Math.PI * .5), false);
+
+			// Bottom side + bottom left corner
+			cornerPath.AddLineToPoint((float)(insetBounds.X + bottomLeft), insetBounds.Height + insetBounds.Y);
+			cornerPath.AddArc((float)(insetBounds.X + bottomLeft), (float)(insetBounds.Y + insetBounds.Height - bottomLeft), (float)bottomLeft, (float)(Math.PI * .5), (float)Math.PI, false);
+
+			// Left side + top left corner
+			cornerPath.AddLineToPoint(insetBounds.X, (float)(insetBounds.Y + topLeft));
+			cornerPath.AddArc((float)(insetBounds.X + topLeft), (float)(insetBounds.Y + topLeft), (float)topLeft, (float)Math.PI, (float)(Math.PI * 1.5), false);
+
+			return cornerPath;
+		}
+
+		public void AddLayer(CALayer layer, int position, UIView viewToAddTo)
+		{
+			// If there is already a layer with the given name, remove it before inserting.
+			if (viewToAddTo.Layer.Sublayers != null && layer != null)
```

Comment:
should go in even when sublayers are null. Means it is the first time its adding layer

Reply:
Fixed. That was a remnant from some layering rework I did, similar to #33.